openssl/src/ssl/test.rs

```diff
@@ -1353,6 +1353,44 @@ fn no_version_overlap() {
     guard.join().unwrap();
 }
 
+#[test]
+#[cfg(all(feature = "v111", ossl111))]
+fn custom_extensions() {
+    static FOUND_EXTENSION: AtomicBool = ATOMIC_BOOL_INIT;
+
+    let listener = TcpListener::bind("127.0.0.1:0").unwrap();
+    let addr = listener.local_addr().unwrap();
+
+    let guard = thread::spawn(move || {
+        let stream = listener.accept().unwrap().0;
+        let mut ctx = SslContext::builder(SslMethod::tls()).unwrap();
+        ctx.set_certificate_file(&Path::new("test/cert.pem"), SslFiletype::PEM)
+            .unwrap();
+        ctx.set_private_key_file(&Path::new("test/key.pem"), SslFiletype::PEM)
+            .unwrap();
+        ctx.add_custom_ext(
+            12345, ssl::ExtensionContext::CLIENT_HELLO,
+            |_, _, _| unreachable!(),
```

Comment:
Should we add versions of add_custom_ext that only set one or the other to avoid having to do this kind of thing?

Reply:
It looks like OpenSSL gracefully skips extensions without a callback set, so that is indeed meaningful. On the other hand, it's really easy to write `|_, _, _| Ok(())` or `Ok(None)` if you want that behavior. Is it worth the extra API surface area?

Reply:
We can leave it as-is for now - we'd need a method that sets both anyways.